backend/src/Edna/Orphans.hs

```diff
@@ -6,21 +6,84 @@ module Edna.Orphans () where
 
 import Universum
 
+import qualified Data.CaseInsensitive as CI
+import Data.Swagger (SwaggerType(..), ToParamSchema(..), format, type_)
+import Fmt (Buildable(..), Builder, blockListF, fmt, listF, tupleF)
+import GHC.TypeLits (KnownSymbol, Symbol, symbolVal)
+import Lens.Micro ((?~))
 import RIO (RIO(..))
+import Servant.API (Header)
+import Servant.API.ResponseHeaders (GetHeaders', Headers, getHeaders, getResponse)
 import Servant.Multipart (MultipartForm')
-import Servant.Util.Combinators.Logging (ApiCanLogArg)
-import Servant.Util.Common.Common (ApiHasArgClass(..))
+import Servant.Pagination (AcceptRanges, ContentRange, Ranges)
+import Servant.Util.Combinators.Logging (ApiCanLogArg, ForResponseLog, buildForResponse)
+import Servant.Util.Common.Common (ApiHasArgClass(..), symbolValT)
 
 -- It's also available in @rio-orphans@, but that would add extra dependencies,
 -- so it's simpler to just have one line for now.
 deriving newtype instance MonadCatch (RIO env)
 
 ----------------
--- Logging
+-- Logging (servant-util)
 ----------------
 
 instance ApiHasArgClass (MultipartForm' mods tag t) where
   type ApiArg (MultipartForm' mods tag t) = t
   apiArgName _ = "multipart"
 
 instance ApiCanLogArg (MultipartForm' mods tag t)
+
+-- TODO [EDNA-87] Instances like this one should be in @servant-util@.
+instance KnownSymbol sym => ApiHasArgClass (Header sym a) where
+  type ApiArg (Header sym a) = Maybe a
+  apiArgName _ = symbolVal @sym Proxy <> " header"
+
+instance KnownSymbol sym => ApiCanLogArg (Header sym a)
+
+instance Show (Ranges fields resource) => Buildable (Ranges fields resource) where
+  -- It's a bit hard to define something better, so we defer to @show@ for now.
+  build = build . show @String
+
+instance (GetHeaders' headers, Buildable a) => Buildable (Headers headers a) where
+  build headers =
+    build (getResponse headers) <>
+    " with headers\n" <> blockListF (map buildHeader untypedHeaders)
+    where
+      untypedHeaders = getHeaders headers
+
+      -- Headers are usually human-readable so applying @show@ should be ok.
+      buildHeader :: (CI.CI ByteString, ByteString) -> Builder
+      buildHeader (CI.original -> headerName, headerBody) =
+        tupleF (show @String headerName, show @String headerBody)
+
+instance (GetHeaders' headers, Buildable a) => Buildable (ForResponseLog $ Headers headers a) where
+  build = buildForResponse
+
+----------------
+-- Swagger
+----------------
+
+instance ToParamSchema (ContentRange fields resource) where
+  -- TODO SOS HELP NEEDED
+  -- How would you define it? I don't know.
+  toParamSchema Proxy = mempty
```

Comment:
I think this is just a string of specific format?

I see below you use `SwaggerArray`, but from my experience, I think the use case for arrays is different, for instance accepting [multiple query parameters](https://hackage.haskell.org/package/servant-swagger-1.1.10/docs/src/Servant.Swagger.Internal.html#line-300). Single `AcceptRanges` equal to `createdAt, modifiedAt` would also be just a mere `SwaggerString`.

Reply:
This is how it looks in Swagger viewer.

![image](https://user-images.githubusercontent.com/2556409/114725348-18d74f00-9d45-11eb-947a-9e1087b4c34b.png)

My problem is that I want to provide some description, but I don't understand what field of `ParamSchema` I need to use to make it appear in `Description` on the screenshot. Probably I'll make another attempt to figure it out when I get some rest from this task :)

Maybe I'll just provide `SwaggerString` type and call it a day because [Content-Range](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range) seems to be a standard header anyway.

Reply:
Yeah, I remember this problem with swagger :fuuu:

In `servant-util` I sort of resolved this with `ParamDescription` type family that associates a type-level description with the given type (e.g. `ContentRange`) + `SwagerrizeApi` that updates descriptions in most places basing on `ParamDescription`, though this is a fairly big hack.